### PR TITLE
Document use of non-default C++ compilers

### LINKED ISSUE #692: Non-default compilers (CMAKE_CXX_COMPILER)
Hi,
I searched the documentation/codebase but could not find an answer: Is it currently possible to use non-gcc/default compilers by setting an option? E.g. if I want a package to build with clang instead of gcc (`CMAKE_C_COMPILER=clang`, `CMAKE_CXX_COMPILER=clang++`)?

Many thanks for your insights :-)

### PR DESCRIPTION
**Pull request by wxmerkt**
Resolves #692

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -173,6 +173,7 @@ Note that some of these currently tied only to a single option, but we still lea
...
 * **CATKIN_LINT_ARGS** (default: not set): If true, you can pass whatever argument(s) ``catkin_lint`` takes, except ``--explain`` that is set by default. Options can be delimit by space if passing multiple.
 * **CMAKE_ARGS** (default: not set): CMake arguments that get passed to the builder for all workspaces.
+* **CC** / **CXX** (default: not set): Environment variables to specify the C/C++ compilers. If required, these can be installed by specifying `ADDITIONAL_DEBS`. E.g. `ADDITIONAL_DEBS=clang CC=clang CXX=clang++` uses Clang to build the workspaces.
```
Comment (gavanderhoorn) [inline doc/index.rst:176]: Would perhaps be good to mention these aren't specific to `industrial_ci`, but the regular env vars you'd use to specify a compiler to use when invoking CMake?

Comment (wxmerkt) [reply doc/index.rst]: Good call - I've just amended. Please let me know if you'd like me to use a different formulation or move it to a separate section

File: doc/index.rst [hunk 2]
```diff
@@ -172,8 +172,9 @@ Note that some of these currently tied only to a single option, but we still lea
...
 * **CATKIN_LINT_ARGS** (default: not set): If true, you can pass whatever argument(s) ``catkin_lint`` takes, except ``--explain`` that is set by default. Options can be delimit by space if passing multiple.
-* **CMAKE_ARGS** (default: not set): CMake arguments that get passed to the builder for all workspaces.
+* **CC** / **CXX** (default: not set): Environment variables to specify the C/C++ compilers. If required, these can be installed by specifying `ADDITIONAL_DEBS`. E.g. `ADDITIONAL_DEBS=clang CC=clang CXX=clang++` uses Clang to build the workspaces. Note, these are the regular environment variables - and they work as pass-through in `industrial_ci` as well.
```
Comment (ipa-mdl) [inline doc/index.rst:175]:
```suggestion
* **CC** / **CXX** (default: not set): Environment variables to specify the C/C++ compilers. If required, these can be installed by specifying ``ADDITIONAL_DEBS``. E.g. ``ADDITIONAL_DEBS=clang CC=clang CXX=clang++`` uses Clang to build the workspaces. Note, these are the regular environment variables - and they work as pass-through in ``industrial_ci`` as well.
```

reStructuredText uses double backticks for preformatted text.

**Comment by ipa-mdl** [comment]
Tests are broken because of https://github.com/ros/rosdistro/pull/29845

**Pull request merged**